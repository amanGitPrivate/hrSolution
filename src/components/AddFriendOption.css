:local(.addFriendRadio) {
    border-radius: 0;
    border-color: #ABAAAA;
    border-left: 0;
    border-right: 0;
  }

:local(.addFriendRadioMaleOption) {
    padding-right:10px;
}