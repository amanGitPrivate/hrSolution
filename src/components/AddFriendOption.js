import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { FEMALE, MALE } from './../constants/GenderConfig';

import styles from './AddFriendOption.css';

const AddFriendOption = props => {
  const { gender, onChange } = props;
  return (
    <div className={classnames('form-control', 'radio', styles.addFriendRadio)}>
      <label className={classnames(styles.addFriendRadioMaleOption)}>
        <input
          type="radio"
          name="gender"
          value={MALE}
          checked={gender === MALE ? true : false}
          onChange={onChange}
        />{' '}
        {MALE}
      </label>
      <label>
        <input
          type="radio"
          name="gender"
          value={FEMALE}
          checked={gender === FEMALE ? true : false}
          onChange={onChange}
        />{' '}
        {FEMALE}{' '}
      </label>
    </div>
  );
};

AddFriendOption.propTypes = {
  gender: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default AddFriendOption;
