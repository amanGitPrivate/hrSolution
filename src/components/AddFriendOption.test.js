import React from 'react';
import AddFriendOption from './AddFriendOption';
import renderer from 'react-test-renderer';

test('AddFriendOption renders correctly', () => {
  const component = renderer.create(
    <AddFriendOption gender={'male'} onChange={jest.fn()} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
