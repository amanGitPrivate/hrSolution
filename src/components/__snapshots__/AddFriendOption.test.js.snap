exports[`test AddFriendOption renders correctly 1`] = `
<div
  className="form-control radio">
  <label
    className="">
    <input
      checked={true}
      name="gender"
      onChange={[Function]}
      type="radio"
      value="male" />
     
    male
  </label>
  <label>
    <input
      checked={false}
      name="gender"
      onChange={[Function]}
      type="radio"
      value="female" />
     
    female
     
  </label>
</div>
`;
